File: algorithms/1002.find-common-characters/solution.rs
impl Solution {
    pub fn common_chars(a: Vec<String>) -> Vec<String> {
        use std::collections::HashMap;

        let mut count_map: HashMap<char, usize> = HashMap::new();

        for c in a[0].chars() {
            let count = count_map.entry(c).or_insert(0);
            *count += 1;
        }

        for i in 1..a.len() {
            let word = &a[i];
            for (key, value) in count_map.iter_mut() {
                let count = word.chars()
                                .collect::<Vec<char>>()
                                .iter()
                                .filter(|&x| x == key)
                                .count();

                *value = std::cmp::min(*value, count);
            }
        }

        count_map
            .iter()
            .flat_map(|(key, value)| std::iter::repeat(key).take(*value))
            .map(|c| c.to_string())
            .collect()
    }
}

File: algorithms/1002.find-common-characters/solution.rb
# @param {String[]} a
# @return {String[]}
def common_chars(a)
  count_map = Hash.new(0)
  a[0].each_char { |c| count_map[c] += 1 }

  (1...a.size).each do |i|
    word = a[i]
    count_map.keys.each { |k| count_map[k] = [count_map[k], word.count(k)].min }
  end

  count_map.map { |k, v| k * v }.join.split('')
end
